eng/common/TestResources/New-TestResources.ps1.md

```diff
@@ -170,97 +171,102 @@ Accept pipeline input: False
+timestamp of the form "2020-03-04T09:07:04.3083910Z".
+
+If this value is not specified no 'DeleteAfter' tag will be assigned to the
+created resource group.
+
+An optional cleanup process can delete resource groups whose "DeleteAfter"
+timestamp is less than the current time.
 
-For more information on the relationship between AAD Applications and Service
-Principals see: https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals
+This isused for CI automation.
 
 ```yaml
-Type: String
+Type: Int32
 Parameter Sets: (All)
 Aliases:
 
 Required: False
 Position: Named
-Default value: None
+Default value: 0
 Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
-### -TenantId
-The tenant ID of a service principal when a provisioner is specified.
-The same
-Tenant ID is used for Test Application and Provisioner Application.
-This value
-is passed to the ARM template as 'tenantId'.
+### -Location
+Optional location where resources should be created.
+If left empty, the default
+is based on the cloud to which the template is being deployed:
+
+* AzureCloud -\> 'westus2'
+* AzureUSGovernment -\> 'usgovvirginia'
+* AzureChinaCloud -\> 'chinaeast2'
 
 ```yaml
 Type: String
-Parameter Sets: Provisioner
+Parameter Sets: (All)
 Aliases:
 
-Required: True
+Required: False
 Position: Named
 Default value: None
 Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
-### -SubscriptionId
-Optional subscription ID to use for new resources when logging in as a
-provisioner.
-You can also use Set-AzContext if not provisioning.
+### -Environment
+Name of the cloud environment.
+The default is the Azure Public Cloud
+('PublicCloud')
 
 ```yaml
 Type: String
-Parameter Sets: Provisioner
+Parameter Sets: (All)
 Aliases:
 
 Required: False
 Position: Named
-Default value: None
+Default value: AzureCloud
 Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
-### -ProvisionerApplicationId
-The AAD Application ID used to provision test resources when a provisioner is
-specified.
-
-If none is specified New-TestResources.ps1 uses the TestApplicationId.
-
-This value is not passed to the ARM template.
+### -AdditionalParameters
+Optional key-value pairs of parameters to pass to the ARM template(s).
 
 ```yaml
-Type: String
-Parameter Sets: Provisioner
+Type: Hashtable
+Parameter Sets: (All)
 Aliases:
 
-Required: True
+Required: False
 Position: Named
 Default value: None
 Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
-### -ProvisionerApplicationSecret
-A service principal secret (password) used to provision test resources when a
-provisioner is specified.
-
-If none is specified New-TestResources.ps1 uses the TestApplicationSecret.
-
-This value is not passed to the ARM template.
+### -SubscriptionConfiguration
+Name of a subscription configuration secret in a Key Vault.
+Stored as a JSON
+object with the expected properties:
+    * SubscriptionId
```

Comment:
This isn't going to be a passed to the ps1 script as a string any longer but it is still worth documenting the format we expect to find in the keyvault, probably in the yml files. I'd add something like:
```
{
"SubscriptionId": "<subscription id>",
"TenantId": "<tenant id>",
"TestApplicationId": "<test app id>",
"TestApplicationSecret": "<test app secret>",
"ProvisionerApplicationId": "<provisoner app id>",
"ProvisoinerApplicationSecret": "<provisoner app secert>",
"Environment": "AzureCloud | AzureGov | AzureChina"
}
```

For the consuming side I expect something like:

```
$subParameters = "${{ parameters.SubParameters }}" | ConvertFrom-Json -AsHashTable
New-TestResources.ps1 `
    -BaseName 'Generated' `
    -ServiceDirectory '$(ServiceDirectory)' `
    -DeleteAfterHours 24 `
    -AdditionalParameters @{ additionalParam1 = 'value'; additionalParam2 = 'value' } `
    -CI `
    -Force `
    -Verbose `
    @subParameters
```